framework/include/base/MooseFunctor.h

```diff
@@ -901,6 +906,7 @@ class FunctorEnvelopeBase
   virtual void jacobianSetup() = 0;
   virtual bool wrapsNull() const = 0;
   virtual std::string returnType() const = 0;
+  virtual bool isConstant() const = 0;
```

Comment:
Yes we do. `FunctorEnvelopeBase` is what allows us to store functors returning all different kinds of types in a single container in `SubProblem`. And because we store them on `SubProblem` as `FunctorEnvelopeBase` we must have the virtual methods on this base class

Reply:
oh smart.